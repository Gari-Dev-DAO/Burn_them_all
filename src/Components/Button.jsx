

const Button = ({children,onClick}) => {
  return (
    <button onClick={onClick} style={{backgroundColor:'transparent',border:'none',cursor:'pointer',zIndex:200}} data-testid='button'>{children}</button>
  )
}

export default Button
