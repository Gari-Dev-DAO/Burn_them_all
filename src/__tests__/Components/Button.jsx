import { fireEvent,render } from "@testing-library/react";
import Button from "../../Components/Button";

describe("Button", () => {
  it("renders component", () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it("Onclick is working", () => {
    const onClick = jest.fn();
    const { getByText } = render(<Button onClick={onClick}>Bonk</Button>);
    const button = getByText("Bonk");
    fireEvent.click(button);
    expect(onClick).toBeCalledTimes(1);
  });
});
