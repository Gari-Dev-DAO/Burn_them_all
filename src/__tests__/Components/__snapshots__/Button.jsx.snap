// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders component 1`] = `
<div>
  <button
    data-testid="button"
    style="background-color: transparent; cursor: pointer; z-index: 200;"
  />
</div>
`;
